### PR TITLE
Polynomial: add perp TVL

### PR DESCRIPTION
**Pull request by gdeschryver**
Using polynomial API to fetch perp margin + smart wallet balances.

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at projects/polynomial: 
        
 
  
    
    
    Error: invalid BigNumber string (argument="value", value="2.720837704928221880193562e+24", code=INVALID_ARGUMENT, version=bignumber/5.7.0)
    Truncated error stack:
        at BigNumber.from (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@ethersproject/bignumber/lib/bignumber.js:199:27)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:334:14
        at Array.map (<anonymous>)
        at computeTVL (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:328:25)
        at getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:54:30)

File: projects/polynomial/index.js [hunk 1]
```diff
@@ -7,6 +7,12 @@ const chain = 'optimism'
 const { getConfig } = require('../helper/cache')
 const transform = t => `${chain}:${t}`
 
+// api
+const BASE_URL = "https://perps-api-experimental.polynomial.fi/snx-perps/tvl";
+const api = axios.create({
```
Comment (g1nt0ki) [inline projects/polynomial/index.js:12]: can this somehow be be fetched on chain?

Comment (gdeschryver) [reply projects/polynomial/index.js:12]: Each user has a Smart Wallet.

First you'd need to get all of them:
https://optimistic.etherscan.io/address/0xd567e18fdf8afa58953dd8b0c1b6c97adf67566b#readContract
accountAddr(id) up to id = accounts (currently 6k+)

Then for each wallet check sUSD balance and how much margin is available on every perp market (currently 24) for this wallet

**Comment by llamatester** [comment]
The adapter at projects/polynomial exports TVL: 
        
 
 
    optimism                  2.92 M
    
    total                    2.92 M

**Comment by llamatester** [comment]
The adapter at projects/polynomial exports TVL: 
        
 
 
    optimism                  2.92 M
    
    total                    2.92 M

**Comment by llamatester** [comment]
The adapter at projects/stellarx exports TVL: 
        
 
 
    
    total                    13.90 M

**Comment by llamatester** [comment]
Error while running adapter at projects/polynomial: 
        
 
  
    
    
    Error: invalid BigNumber string (argument="value", value="2.800184684554827686762199e+24", code=INVALID_ARGUMENT, version=bignumber/5.7.0)
    Truncated error stack:
        at BigNumber.from (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@ethersproject/bignumber/lib/bignumber.js:199:27)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:334:14
        at Array.map (<anonymous>)
        at computeTVL (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:328:25)
        at getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:54:30)

**Comment by g1nt0ki** [comment]
@gdeschryver the stellarx commit was pushed to wrong branch, but not sure why you force pushed your changes, because now my bugfix to your adapter is gone

**Comment by llamatester** [comment]
The adapter at projects/polynomial exports TVL: 
        
 
 
    optimism                  3.00 M
    
    total                    3.00 M

**Comment by gdeschryver** [comment]
> @gdeschryver the stellarx commit was pushed to wrong branch, but not sure why you force pushed your changes, because now my bugfix to your adapter is gone

My bad, got confused by the stellarx commit, tought your polynomial fix was merged on main and I tried to rebase.
Added your fix back with last commit, thanks !

**Comment by g1nt0ki** [comment]
@gdeschryver all good, thanks for the PR

**Pull request merged**